src/plugins/image/src/main/js/ui/Dialog.js

```diff
@@ -61,12 +61,16 @@ define(
         }
 
         src = imgElm ? dom.getAttrib(imgElm, 'src') : '';
+        src = Utils.getInternalImageSrc(editor, src);
 
         // Leave the chooser empty if the URL is external (ie doesn't start with site:// or /).
         if (!Utils.isInternalUrl(src)) {
           src = '';
         }
 
+        // Workaround to ensure any bad internal src is updated
+        dom.setAttrib(imgElm, 'src', src);
```

Comment:
For some reason, setting the src again here works. I used this same line in the filter callback, as well as trying to replace that node and other experiments, none I was able to get to work yet 👎